system/lib/libc/musl/src/exit/abort.c

```diff
@@ -6,8 +6,15 @@
 #include "lock.h"
 #include "ksigaction.h"
 
+#if __EMSCRIPTEN__
+_Noreturn void _abort_js();
+#endif
+
 _Noreturn void abort(void)
 {
+#if __EMSCRIPTEN__
+	_abort_js();
+#else
```

Comment:
What's the advantage of including this `.c` file if it just forwards out to JS? Implementing it in JS directly seems simpler.

Reply:
Sorry I should have mentioned, this was needed as part of an experiment I'm doing to remove the underscore name mangling.   One place where that is an issue is where the new library function name `abort` vs `_abort` clashed with some existing JS symbol, in this case the JS abort utility defined in the preamble.

Avoiding these clashing names seems like a good thing to avoid confusion anyway.. since if you read the current JS abort library function it looks like its infinite recursion, when in fact its calling from the C-facing abort to the JS-flavored abort (that later takes a string argument).

Reply:
I see. Yeah, as a workaround for that problem this sounds reasonable (with a comment in the source maybe saying that is why it is needed).

Reply:
Can you also mention this in the commit description?